Code/GraphMol/WedgeBonds.cpp

```diff
@@ -0,0 +1,493 @@
+    if (bond->getBondType() != Bond::SINGLE) {
+      continue;
+    }
+
+    int bid = bond->getIdx();
+    if (resSoFar.find(bid) == resSoFar.end()) {
+      // very strong preference for Hs:
+      if (bond->getOtherAtom(atom)->getAtomicNum() == 1) {
+        nbrScores.emplace_back(-1000000,
+                               bid);  // lower than anything else can be
+        continue;
+      }
+      // prefer lower atomic numbers with lower degrees and no specified
+      // chirality:
+      auto *oatom = bond->getOtherAtom(atom);
+      int nbrScore = oatom->getAtomicNum() + 100 * oatom->getDegree() +
+                     1000 * ((oatom->getChiralTag() != Atom::CHI_UNSPECIFIED));
+      // prefer neighbors that are nonchiral or have as few chiral neighbors
+      // as possible:
+      int oIdx = oatom->getIdx();
+      if (nChiralNbrs[oIdx] < noNbrs) {
+        // the counts are negative, so we have to subtract them off
+        nbrScore -= 100000 * nChiralNbrs[oIdx];
+      }
+      // prefer bonds to non-ring atoms:
+      nbrScore += 10000 * mol.getRingInfo()->numAtomRings(oIdx);
+      // prefer non-ring bonds;
+      nbrScore += 20000 * mol.getRingInfo()->numBondRings(bid);
+      // prefer bonds to atoms which don't have a double bond from them
+      auto [hasDoubleBond, hasKnownDoubleBond, hasAnyDoubleBond] =
+          getDoubleBondPresence(mol, *oatom);
+      nbrScore += 11000 * hasDoubleBond;
+      nbrScore += 12000 * hasKnownDoubleBond;
+      nbrScore += 23000 * hasAnyDoubleBond;
+
+      // std::cerr << "    nrbScore: " << idx << " - " << oIdx << " : "
+      //           << nbrScore << " nChiralNbrs: " << nChiralNbrs[oIdx]
+      //           << std::endl;
+      nbrScores.emplace_back(nbrScore, bid);
+    }
+  }
+  // There's still one situation where this whole thing can fail: an unlucky
+  // situation where all neighbors of all neighbors of an atom are chiral and
+  // that atom ends up being the last one picked for stereochem assignment. This
+  // also happens in cases where the chiral atom doesn't have all of its
+  // neighbors (like when working with partially sanitized fragments)
+  //
+  // We'll bail here by returning -1
+  if (nbrScores.empty()) {
+    return -1;
+  }
+  auto minPr = std::min_element(nbrScores.begin(), nbrScores.end());
+  return minPr->second;
+}
+
+}  // namespace detail
+
+// returns map of bondIdx -> bond begin atom for those bonds that
+// need wedging.
+INT_MAP_INT pickBondsToWedge(const ROMol &mol,
+                             const BondWedgingParameters *params) {
+  if (!params) {
+    params = &defaultWedgingParams;
+  }
+  std::vector<unsigned int> indices(mol.getNumAtoms());
+  std::iota(indices.begin(), indices.end(), 0);
+  static int noNbrs = 100;
+  auto [chiNbrs, nChiralNbrs] = detail::countChiralNbrs(mol, noNbrs);
+  if (chiNbrs) {
+    std::sort(indices.begin(), indices.end(), [&](auto i1, auto i2) {
+      return nChiralNbrs[i1] < nChiralNbrs[i2];
```

Comment:
You're going to love this one:

https://stackoverflow.com/questions/46114214/lambda-implicit-capture-fails-with-variable-declared-from-structured-binding

Basically clang is being anal with the specification that tuple unpacking never makes variables and lambdas must only implicitly bind variables.

Reply:
I believe you can resolve this as follows:

```
std::sort(indices.begin(), indices.end(), [&, nChiralNbrs=nChiralNbrs](auto i1, auto i2) {
       return nChiralNbrs[i1] < nChiralNbrs[i2];
});
```

Reply:
Thanks for pointing that out.